docs/android/README.md

```diff
@@ -0,0 +1,119 @@
+GradleSourceSet model, already used for Java projects, to store and manage Android build variant information. Here's a
+breakdown of the magic behind the scenes:
+
+### 1. Project Detection:
+
+Gradle Build Server detects if the Android plugin is applied to the project. It then uses the appropriate model
+builders to handle both Java and Android specifics.
+
+![SourceSet build logic](../../images/sourceset_build_logic.png)
+
+### 2. Extracting Information:
+
+We leverage reflection to gather necessary details about the project, such as compile tasks and dependencies. All the
+extracted data is used to build a `GradleSourceSet` model.
+
+- We map Android variant's `assemble` task to the java sourceset's `classes` task, which ensures all resources and
+  source code are compiled and packaged. This assemble task name follows the format `assemble<variant_name>`.
+  E.g. `assembleRelease` for `release` variant and `assembleDebug` for `debug` variant.
+- Java compile task responsible for compiling all the java code for the build variant follows a similar task name
+  format `process<variant_name>JavaWithJavac`. E.g. `processReleaseJavaWithJavac` for `release` variant
+  and `processDebugJavaWithJavac` for `debug` variant.
+- Similar to Java projects, the `AndroidDependencyCollector` class handles module dependencies, including R files
+  and Android SDK components.
+- A build variant can have multiple source sets and each source set has java directories. We collect the list of
+  these java directories which serve as the source directories.
+- For the resource directories we take a similar approach of collecting the res and resource directories of the
+  build variant's source sets.
+- `processJavaResources` and `mergeResources` task provide the output directories of java and android resources.
+- Using `javaCompile` task we fetch destination directory, compiler source directories and compiler args where the
+  destination directory is used as `sourceOutputDirs`. Next we remove all the source directories from the compiler
+  source directories to get the `generatedSourceDirs`.
+- Classpath files for the variant is retrieved from the `compileConfiguration`.
+- To determine if the variant has tests we check if android or unit tests are present in the given variant.
+- Finally, we create the java extension using the compiler args we fetched earlier.
+
+This integration allows Gradle Build Server to handle your Android projects just like any other Java or Scala project,
+ensuring a familiar and efficient workflow.
+
+> [!NOTE]
+> This is an ongoing effort, and future updates will expand Android support.
+
+## Android Build Process: Relevant Part
+
+The following diagram showcasing Android build process has been abstracted down to only show the parts which is relevant
+to this project's implementation.
+
+![Android Build Process](../../images/android_build_process.png)
```

Comment:
It depends on the documentation engine, but keeping such content under `docs/images` or event `docs/android/images` looks more convenient

Reply:
Addressed in https://github.com/microsoft/build-server-for-gradle/pull/189/commits/b624852d90f8acc723f4bb12128d95648ac18442